evap/staff/views.py

```diff
@@ -276,41 +276,51 @@ def warning_for_inapplicables(amount):
 
     @staticmethod
     def apply(
-        request, evaluations, email_template=None, email_template_contributor=None, email_template_participant=None
+        request,
+        evaluations,
+        email_template=None,
+        email_template_contributor=None,
+        email_template_participant=None,
+        delete_previous_answers=None,
     ):
         raise NotImplementedError
 
 
-class RevertToNewOperation(EvaluationOperation):
-    confirmation_message = gettext_lazy("Do you want to revert the following evaluations to preparation?")
+class ResetToNewOperation(EvaluationOperation):
+    confirmation_message = gettext_lazy("Do you want to reset the following evaluations to preparation?")
 
     @staticmethod
-    def applicable_to(evaluation):
-        return Evaluation.State.PREPARED <= evaluation.state <= Evaluation.State.APPROVED
+    def applicable_to(evaluation: Evaluation):
+        return evaluation.can_reset_to_new()
 
     @staticmethod
     def warning_for_inapplicables(amount):
         return ngettext(
-            "{} evaluation cannot be reverted, because it already started. It was removed from the selection.",
-            "{} evaluations cannot be reverted, because they already started. They were removed from the selection.",
+            "{} evaluation cannot be reset, because it is already published. It was removed from the selection",
+            "{} evaluations cannot be reset, because they were already published. They were removed from the selection.",
             amount,
         ).format(amount)
 
     @staticmethod
     def apply(
-        request, evaluations, email_template=None, email_template_contributor=None, email_template_participant=None
+        request,
+        evaluations,
+        email_template=None,
+        email_template_contributor=None,
+        email_template_participant=None,
+        delete_previous_answers=None,
     ):
         assert email_template_contributor is None
         assert email_template_participant is None
 
         for evaluation in evaluations:
-            evaluation.revert_to_new()
+            evaluation.reset_to_new(delete_previous_answers=delete_previous_answers)
```

Comment:
as noted, it is a BIT sus that we pass `bool | None` to `reset_to_new`, but I think it's okayish here

Reply:
I'd prefer the narrower contract on the function, i.e. annotating the function as only taking in a `bool` and then explicitly casting the possibly None value to bool on the caller site. What do you think about that?

Reply:
Yeah, we can just do it narrower. About casting on the caller site:

The only caller site is currently the `ResetToNewOperation`. We could do it like:

```python
evaluation.reset_to_new(delete_previous_answers=bool(delete_previous_answers))
```

But as `delete_previous_answers` is not currently typed inside `ResetToNewOperation.apply()`, nothing complains if we still just do:
```python
evaluation.reset_to_new(delete_previous_answers=delete_previous_answers)
```

Reply:
@richardebeling @Kakadus I completely removed the inheritance and instead use types to validate that the correct method is called.

I am also thinking of adding a test like 
```python
class TestWrongOperationArgs(TestCase):
    def test_wrong_args_throw_typeerror(self):
        evaluation = baker.make(
            Evaluation,
            state=Evaluation.State.PREPARED
        )
        _, operation = evap.staff.views.target_state_and_operation_from_str(str(Evaluation.State.NEW))

        args = {"email_template": EmailTemplate()}

        self.assertRaisesRegex(
            TypeError,
            "got an unexpected keyword argument",
            operation.apply,
            Mock(), [evaluation], **args
        )
```

Do you think this makes sense? Is this not needed? Or should I cover more/all of the operations?